src/App.tsx

```diff
@@ -1,7 +1,37 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type AppState = {
+  key: string;
+};
+
+export class App extends React.Component {
+  state: AppState = {
+    key: '',
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleKeyup);
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', this.handleKeyup);
+  }
+
+  handleKeyup = (e: KeyboardEvent) => {
+    this.setState({ key: e.key });
+  };
```

Comment:
you can use destructuring here
```suggestion
  handleKeyup = ({ key }: KeyboardEvent) => {
    this.setState({ key });
  };
```